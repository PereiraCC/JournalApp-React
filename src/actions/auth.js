import { googleAuthProvider, getAuth, signInWithPopup, createUserWithEmailAndPassword, updateProfile, signInWithEmailAndPassword } from '../firebase/firebase-config';
import { types } from '../types/types';
import { finishLoading, startLoading } from './ui';

export const startLoginEmailPassword = (email, password) => {
    return ( dispatch ) => {

        dispatch( startLoading() );

        const auth = getAuth();
        signInWithEmailAndPassword(auth, email, password)
            .then(({ user }) => {
                dispatch( login( user.uid, user.displayName ) );
                
                dispatch( finishLoading() );
            })
            .catch( e => {
                dispatch( finishLoading() );
                console.log(e);
            });      
    }
}

export const startRegisterWithEmailPasswordName = (email, password, name) => {

    return (dispatch ) => {
        const auth = getAuth();
        createUserWithEmailAndPassword(auth, email, password )
            .then( async ({ user }) => {

                await updateProfile( auth.currentUser, { 
                    displayName: name
                });

                dispatch(
                    login( user.uid, user.displayName )
                );
            })
            .catch( e => {
                console.log(e);
            })
    }

}

export const startGoogleLogin = () => {
    return ( dispatch ) => {

        const auth = getAuth();
        signInWithPopup(auth, googleAuthProvider)
            .then( ({ user }) => {
                dispatch(
                    login( user.uid, user.displayName )
                )
            });
    }
}

export const login = ( uid, displayName ) => ({
    type: types.login,
    payload: {
        uid,
        displayName
    }
});

// export const logout = () => ({
//     type: types.logout,
// });



