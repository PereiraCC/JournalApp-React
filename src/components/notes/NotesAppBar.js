import { useDispatch, useSelector } from 'react-redux';

import { startSaveNote, startUploading } from '../../actions/notes';

export const NotesAppBar = () => {

    const { active } = useSelector( state => state.notes );
    const dispatch = useDispatch();

    const handleSaveNote = () => {
        dispatch( startSaveNote( active ) );
    }

    const handleUploadPicture = () => {
        document.querySelector('#fileSelector').click();
    }

    const handleFileChange = (e) => {
        const file = e.target.files[0];
        if( file ) {
            dispatch( startUploading( file ) );
        }
    }

    return (
        <div className="notes__appbar">
            <span>29 January 2022</span>

            <input 
                id="fileSelector"
                type="file"
                name="file"
                style={ { display: 'none' } }
                onChange={ handleFileChange }
            />

            <div>
                <button 
                    className="btn"
                    onClick={ handleUploadPicture }    
                >
                    Picture
                </button>
                
                <button 
                    className="btn"
                    onClick={ handleSaveNote }
                >
                    Save
                </button>
            </div>
        </div>
    );
};
