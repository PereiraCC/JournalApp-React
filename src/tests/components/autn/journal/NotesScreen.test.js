import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { activeNote } from '../../../../actions/notes';
import { NoteScreen } from '../../../../components/notes/NoteScreen';

jest.mock('../../../../actions/notes', () => ({
    activeNote: jest.fn()
}));

const middlewares = [ thunk ];
const mockStore = configureStore(middlewares);

const initialState = {
    auth: {
        uid: '1',
        name: 'Carlos',
    },
    ui: {
        loading: false,
        msgError: null,
    },
    notes: {
        active: {
            id: 1234,
            title: 'The best title',
            body: 'The best body',
            date: 0
        },
        notes: []
    }
};

let store = mockStore( initialState );
store.dispatch = jest.fn();

const wrapper = mount( 
    <Provider store={ store} >
        <NoteScreen/>
    </Provider>
);

describe('<Sidebar /> test', () => { 

    test('should show correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should triger the active note', () => { 

        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title',
                value: 'The best title',
            }
        });

        expect( activeNote ).toHaveBeenLastCalledWith(
            1234,
            {
                body: 'The best body',
                title: 'The best title',
                id: 1234,
                date: 0
            }
        );

    });

});