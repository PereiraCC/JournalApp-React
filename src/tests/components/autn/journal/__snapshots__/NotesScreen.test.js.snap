// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> test should show correctly 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      29 January 2022
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesone title"
      type="text"
      value="The best title"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today"
      value="The best body"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
